10- Dad Jokes/script.js

```diff
@@ -0,0 +1,57 @@
+const jokeEl = document.getElementById('joke')
+const jokeBtn = document.getElementById('jokeBtn')
+
+jokeBtn.addEventListener('click', generateJoke)
```

Comment:
Question: What happens if you keep on clicking the button?
Suggestion: I think you need to add mechanism to know if we are currently fetching a joke, so that we can either disable the button or ignore clicks, What do you think?

ex.
```
function generateJoke() {
 jokeBtn.disabled = true;  
 ...
 ...
  axios.get(URL, CONFIG).then((res) => {
    ...
  }).finally(() => {
    jokeBtn.disabled = false;  
});
}
```

Reply:
1- if you want my opinion : i dont really think that a problem beacuse even if i keept clicking the button i can still see the next joke and i dont really feel it's a problem

2 i did try to add a what you said i couldn't really make it but this what i came out with as a temporary solution
```suggestion
jokeBtn.addEventListener('click', generateJoke)
function submitBtn() {
  document.getElementById("jokeBtn").disabled = true;
  setTimeout(function () { document.getElementById("jokeBtn").disabled = false; }, 500);
}

generateJoke()

// ********** USING AXIOS **********

function generateJoke() {
  const config = {
    headers: {
      'Accept': 'application/json'
    }
  }

  axios.get('https://icanhazdadjoke.com', config).then((res) => {
    jokeEl.textContent = res.data.joke
  })
  submitBtn()
}

```